### PR TITLE
Update to newer dependencies

### PR DESCRIPTION
**Pull request by reynir**
Mirage-crypto>= 1.0.0, dream.1.0.0~alpha7 and get rid of cstruct.

I will need to test it (which may take some time as it requires https)

### CODE REVIEW

**Comment by hannesm** [comment]
you/we can setup a forward from tlstunnel for *.robur.coop and then test/run the demo again!?

File: webauthn.opam [hunk 1]
```diff
@@ -16,20 +16,21 @@ build: [
...
+  "digestif"
+  "mirage-crypto-ec" {>= "1.1.0"}
+  "mirage-crypto-rng" {>= "1.1.0"}
   "ocplib-endian"
-  "x509" {>= "0.13.0"}
+  "x509" {>= "1.0.2"}
   "base64" {>= "3.1.0"}
-  "cstruct" {>= "6.0.0"}
   "cbor" {>= "0.5"}
+  "ohex"
```
Comment (hannesm) [inline webauthn.opam:33]:
```suggestion
  "ohex" {>= "0.2.0"}
```

**Comment by hannesm** [comment]
looks good to me

**Comment by reynir** [comment]
It seems to work fine! I deployed it on webauthn-demo.robur.coop. I still have to write a rc script to run the executabe automatically.

**Pull request merged**